### PR TITLE
Update version of Patternfly to v6

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Working on Spike: [RHOAIENG-9577](https://issues.redhat.com/browse/RHOAIENG-9577)

Steps:

***Update package.json to latest v6 versions and update the overrides (manually)***
<commit 1>

***Run the package install***
npm i

***Run code mods***
npx @patternfly/pf-codemods@latest frontend/src/ --v6

***Fix Code mods fail causes (manually)***
<commit 2>

***Re-Run code mods***

npx @patternfly/pf-codemods@latest frontend/src/ --v6

npx @patternfly/pf-codemods@latest frontend/src/ --v6 —fix
<commit 3>

***Fix Remaining Code Mod Errors (manually)***
<commit 4>

***Fix Lint Errors***
npm run test:fix 
<commit 5>

***Manually fix remaining Lint Errors (manually)***
<commit 6>

***Run class-name-updater***

npx @patternfly/class-name-updater --v6 frontend/src/ --fix
<commit 7>

***Build attempt***
npm run build

***Fix remaining build issues (manually)***
<commit 8>

***Replace “—pf-v6-global—spacer” with “—pf-t—global—spacer”***
<commit 9>

***Home page fix (manually)***
<commit 10>

image

### CODE REVIEW

File: frontend/src/components/DashboardDescriptionListGroup.tsx [hunk 1]
```diff
@@ -110,7 +109,7 @@ const DashboardDescriptionListGroup: React.FC<DashboardDescriptionListGroupProps
           </Flex>
         </DescriptionListTerm>
       )}
-      <DescriptionListDescription className={isEmpty && !isEditing ? text.disabledColor_100 : ''}>
+      <DescriptionListDescription className={isEmpty && !isEditing ? text.textColorDisabled : ''}>
```
Comment (Griffin-Sullivan) [inline frontend/src/components/DashboardDescriptionListGroup.tsx:112]: Note: you will need to conditionally render `aria-disabled` to get passed the accessbility testing. See https://github.com/kubeflow/model-registry/pull/434

File: frontend/src/app/AboutDialog.scss [hunk 2]
```diff
@@ -3,7 +3,7 @@
 // Override to use the full width of the dialog rather than wrapping early
 .odh-about-dialog {
```
Comment (jenny-s51) [inline frontend/src/app/AboutDialog.scss:4]: [patternfly/patternfly/issues/6871](https://github.com/patternfly/patternfly/issues/6871) has been resolved in v6. Once the fix has been verified on this branch, `AboutDialog.scss` and any related references can be removed as part of [RHOAIENG-9742](https://issues.redhat.com/browse/RHOAIENG-9742).

Comment (jenny-s51) [reply frontend/src/app/AboutDialog.scss:4]: Opened a PR on your branch @jeff-phillips-18 https://github.com/jeff-phillips-18/odh-dashboard/pull/40

File: frontend/src/app/App.scss [hunk 3]
```diff
@@ -7,80 +7,44 @@ body,
...
 
-.pf-v5-c-truncate {
-  --pf-v5-c-truncate--MinWidth: 0;
-  --pf-v5-c-truncate__start--MinWidth: 0;
+.pf-v6-c-truncate {
+  --pf-v6-c-truncate--MinWidth: 0;
+  --pf-v6-c-truncate__start--MinWidth: 0;
+}
+
+// TODO: Remove when PF bug is fixed.
```
Comment (christianvogt) [inline frontend/src/app/App.scss:41]:
```suggestion
// TODO: Remove when PF bug is fixed. https://github.com/patternfly/patternfly-react/issues/11314
```

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelinesList.cy.ts [hunk 4]
```diff
@@ -163,7 +163,8 @@ describe('PipelinesList', () => {
     pipelineRow
       .getPipelineVersionRowById(initialMockPipelineVersion.pipeline_version_id)
       .findPipelineVersionLink()
-      .click();
+      // TODO - remove force one https://github.com/patternfly/patternfly-react/issues/11314 is fixed
+      .click({ force: true });
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelinesList.cy.ts:167]: With the css override for table truncation, I think we can remove this workaround.
```suggestion
      .click();
```

File: frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelines.cy.ts [hunk 5]
```diff
@@ -941,7 +941,7 @@ describe('Pipelines', () => {
     pipelineRow
       .getPipelineVersionRowById(initialMockPipelineVersion.pipeline_version_id)
       .findPipelineVersionLink()
-      .click();
+      .click({ force: true });
```
Comment (christianvogt) [inline frontend/src/__tests__/cypress/cypress/tests/mocked/pipelines/pipelines.cy.ts:944]: With the css override for table truncation, I think we can remove this workaround.
```suggestion
      .click();
```

**Comment by christianvogt** [comment]
Tested many areas and while there's still some outstanding issues I have shared with @jeff-phillips-18 and PF team, I haven't run into any issues that would be considered a breaking issue.

**Comment by jenny-s51** [comment]
Tested many areas throughout the UI as well and noted a few outstanding issues which we agreed to capture in the existing epic as a follow-up story. Overall looks fantastic, really great work here @jeff-phillips-18

**Comment by christianvogt** [comment]
/lgtm
/approve

god speed ('-')7

**Pull request merged**